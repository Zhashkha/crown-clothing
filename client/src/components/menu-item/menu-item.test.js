import React from "react";
import { shallow } from "enzyme";

import { MenuItem } from "./menu-item.component";

describe("test MenuItem", () => {
  let wrapper;
  let mockHistory;
  let mockMatch;
  let mockLinkUrl;

  beforeEach(() => {
    mockHistory = {
      push: jest.fn()
    };
    mockMatch = {
      url: "url"
    };
    mockLinkUrl = "linkUrl";

    const mockProps = {
      title: "",
      imageUrl: "",
      size: "",
      history: mockHistory,
      match: mockMatch,
      linkUrl: mockLinkUrl
    };

    wrapper = shallow(<MenuItem {...mockProps} />);
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call history.push on MenuItemContainer click", () => {
    wrapper.find("MenuItemContainer").simulate("click");
    expect(mockHistory.push).toHaveBeenCalledWith(
      `${mockMatch.url}${mockLinkUrl}`
    );
  });
});
