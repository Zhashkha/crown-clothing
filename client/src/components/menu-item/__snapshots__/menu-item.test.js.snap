// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test MenuItem should match snapshot 1`] = `ShallowWrapper {}`;
